fastlane/Fastfile

```diff
@@ -919,55 +919,103 @@ platform :ios do
+# Optionally, it can create any new necessary certificates or profiles.
+#
+# @option [Boolean] readonly (default: true) Whether to only fetch existing certificates and profiles, without generating new ones.
+#
+private_lane :alpha_code_signing do |options|
+  update_code_signing_enterprise(
+    app_identifiers: simplenote_app_identifiers(root_bundle_id: "#{APP_STORE_BUNDLE_IDENTIFIER}.Alpha"),
+    readonly: options.fetch(:readonly, true)
+  )
+end
+
+# Downloads all the required certificates and profiles (using `match`) for the internal builds in the enterprise account.
+# Optionally, it can create any new necessary certificates or profiles.
+#
+# @option [Boolean] readonly (default: true) Whether to only fetch existing certificates and profiles, without generating new ones.
+#
 private_lane :internal_code_signing do |options|
-  match(
-    type: 'enterprise',
-    team_id: get_required_env('INT_EXPORT_TEAM_ID'),
-    readonly: options[:readonly] || is_ci,
-    app_identifier: simplenote_app_identifiers(root_bundle_id: "#{APP_STORE_BUNDLE_IDENTIFIER}.Internal")
+  update_code_signing_enterprise(
+    app_identifiers: simplenote_app_identifiers(root_bundle_id: "#{APP_STORE_BUNDLE_IDENTIFIER}.Internal"),
+    readonly: options.fetch(:readonly, true)
   )
 end
 
+# Downloads all the required certificates and profiles for the production build.
+# Optionally, it can create any new necessary certificates or profiles.
+#
+# @option [Boolean] readonly (default: true) Whether to only fetch existing certificates and profiles, without generating new ones.
 private_lane :appstore_code_signing do |options|
-  match(
+  update_code_signing(
     type: 'appstore',
     team_id: get_required_env('EXT_EXPORT_TEAM_ID'),
-    readonly: options[:readonly] || is_ci,
-    app_identifier: simplenote_app_identifiers,
-    template_name: 'NotationalFlow Keychain Access (Distribution)'
+    readonly: options.fetch(:readonly, true),
+    app_identifiers: simplenote_app_identifiers,
+    api_key_path: APP_STORE_CONNECT_KEY_PATH
+  )
+end
+
+def update_code_signing(type:, team_id:, readonly:, app_identifiers:, api_key_path:)
+  # NOTE: It might be neccessary to add `force: true` alongside `readonly: true` in order to regenerate some provisioning profiles.
+  # If this turns out to be a hard requirement, we should consider updating the method with logic to toggle the two setting based on whether we're fetching or renewing.
+
+  # Fail early if secrets not available (via ENV.fetch with not default).
+  # Otherwise, Fastlane will prompt to type them.
+  access_key = ENV.fetch('MATCH_S3_ACCESS_KEY')
+  secret_access_key = ENV.fetch('MATCH_S3_SECRET_ACCESS_KEY')
+  # This ENV var is required but match has no input parameter for it, likely for security
+  password_key = 'MATCH_PASSWORD'
+  raise "Missing #{password_key} in the environment" unless ENV.key?(password_key)
```

Comment:
Good point. I removed the extra check here so that there won't be problems if `MATCH_PASSWORD` is provided locally as an env var. https://github.com/Automattic/simplenote-ios/commit/4a23e75e25f2e52d44e6789a90a4df48b64d8685